src/lib/SelectionModel.js

```diff
@@ -107,6 +107,14 @@ SelectionModel.prototype = {
      */
     select: function(ox, oy, ex, ey) {
         var newSelection = this.grid.newRectangle(ox, oy, ex, ey);
+        newSelection.firstSelectedCell = this.grid.newPoint(ox, oy); //Cache the first selected cell before it gets normalized to top-left origin
+        newSelection.lastSelectedCell = (
+            (newSelection.firstSelectedCell.x === newSelection.origin.x && newSelection.firstSelectedCell.y === newSelection.origin.y)
+            ?
```

Comment:
good to go....
however going forward, please let's not put operators on lines by themselves. thx...
My standard for this statement would have been:

```javascript
        newSelection.lastSelectedCell = (
            newSelection.firstSelectedCell.x === newSelection.origin.x &&
            newSelection.firstSelectedCell.y === newSelection.origin.y
        )
            ? newSelection.corner
            : newSelection.origin;
````